### blibliotecas iniciais 

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import warnings
import os
warnings.filterwarnings("ignore")


### Blibliotecas para tratamento de imagem

In [2]:
from skimage.color import rgb2gray
from skimage.filters import try_all_threshold, threshold_otsu, threshold_local, gaussian
from skimage.transform import rotate, rescale
from skimage import morphology
import cv2

In [3]:
# G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Early blight
# G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Healthy
# G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Late blight

In [4]:
trainEarly = os.listdir('G:\\Meu Drive\Renan-Lemes-TCC\\dataset\\Potato\\train\\Potato Early blight')
trainHeal = os.listdir('G:\\Meu Drive\\Renan-Lemes-TCC\\dataset\\Potato\\train\\Potato Healthy')
trainLate = os.listdir('G:\\Meu Drive\\Renan-Lemes-TCC\\dataset\\Potato\\train\\Potato Late blight')


In [5]:
print('-----train-----')
print(len(trainHeal))
print(len(trainEarly))
print(len(trainLate))

-----train-----
2432
3532
3521


### Ai decidi Pegar apenas  as 2432 das saudaveis, e metade das Late e metade Early ou seja 1216 cada.

In [34]:
print(round(len(trainHeal)/2 ))
amostLen = round(len(trainHeal)/2)

1216


In [7]:
path1 = 'G:\\Meu Drive\Renan-Lemes-TCC\\dataset\\Potato\\train\\Potato Early blight' ## não saudavel então pego só 1216
path2 = 'G:\\Meu Drive\\Renan-Lemes-TCC\\dataset\\Potato\\train\\Potato Healthy' ## saudavel pego todas
path3 = 'G:\\Meu Drive\\Renan-Lemes-TCC\\dataset\\Potato\\train\\Potato Late blight' ## não saudavel então pego só 1216

In [35]:
cont = 0
for i in os.listdir(path1):
    cont += 1
    print(path1 + "\\" + i)
    if cont == amostLen:
        break

G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Early blight\PotatoEarlyBlight(3020).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Early blight\PotatoEarlyBlight(2881).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Early blight\PotatoEarlyBlight(2729).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Early blight\PotatoEarlyBlight(3075).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Early blight\PotatoEarlyBlight(3050).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Early blight\PotatoEarlyBlight(3226).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Early blight\PotatoEarlyBlight(3308).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Early blight\PotatoEarlyBlight(3250).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Early blight\PotatoEarlyBlight(3274).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Early blight\PotatoEarlyBlight(3188).JPG
G:\Meu Drive\Renan-L

In [36]:
cont = 0
for i in os.listdir(path3):
    cont += 1
    print(path3 + "\\" + i)
    if cont == amostLen:
        break

G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Late blight\PotatoLateBlight(3451).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Late blight\PotatoLateBlight(3180).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Late blight\PotatoLateBlight(3088).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Late blight\PotatoLateBlight(3487).jpg
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Late blight\PotatoLateBlight(3143).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Late blight\PotatoLateBlight(3229).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Late blight\PotatoLateBlight(3169).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Late blight\PotatoLateBlight(3314).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Late blight\PotatoLateBlight(3267).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Potato\train\Potato Late blight\PotatoLateBlight(3402).JPG
G:\Meu Drive\Renan-Lemes-TCC\dataset\Pot

# Notei que já separou as imagens e quantificou. Porem, ainda para tratar as imagens deixa tudo na mesma pasta inicialmente, somente separando as classes, inclusive porque voce não finalizou os experimentos com a extração de características.

# Agora as etapas de execução:
- Inserir aqui experimentos separando de cada canal de cor. Exemplo exibir a imagem do H, do S e do L e depois imagem do H, do S e do V. Não precisa ser de cada pasta de treino e validação, como voce vai juntar tudo inicialmente, faz 1 experimento de cada classe so. 

- Verificar o canal que mais tem contorno, sem tantas misturas de cores. 
- Realizar a equalização do histograma nesse canal separado
- juntar a imagem novamente para formar nova imagem e retornar para RGB. 

- Após isso, voce irá iniciar a extração de características. 
- indico usar inicialmente como extrator de características o SIFT. 
https://www.dominodatalab.com/blog/feature-extraction-and-image-classification-using-deep-neural-networks 

Somente após isso, siga para os experimentos com aprendizagem de máquina. 
Finalize esses passos acima e me avise para iniciarmos os treinamentos.